File: data/EXP01/python/2257/2257.output_2.py
# @lc app=leetcode id=2257 lang=python3
from typing import List
from collections import deque

class Solution:
    def countUnguarded(self, m: int, n: int, guards: List[List[int]], walls: List[List[int]]) -> int:
        grid = [[0] * n for _ in range(m)]
        
        for wall in walls:
            row, col = wall
            grid[row][col] = -1
        
        for guard in guards:
            row, col = guard
            queue = deque([(row, col)])
            visited = set([(row, col)])
            
            while queue:
                curr_row, curr_col = queue.popleft()
                grid[curr_row][curr_col] = 1
                
                for d_row, d_col in [(-1, 0), (1, 0), (0, -1), (0, 1)]:
                    new_row, new_col = curr_row + d_row, curr_col + d_col
                    
                    if 0 <= new_row < m and 0 <= new_col < n and grid[new_row][new_col] == 0 and (new_row, new_col) not in visited:
                        queue.append((new_row, new_col))
                        visited.add((new_row, new_col))
        
        unguarded_count = sum(row.count(0) for row in grid)
        
        return unguarded_count

File: data/EXP01/go/2257/2257.output_2.go
func countUnguarded(m int, n int, guards [][]int, walls [][]int) int {
    grid := make([][]int, m)
    for i := 0; i < m; i++ {
        grid[i] = make([]int, n)
    }
    
    for _, wall := range walls {
        row, col := wall[0], wall[1]
        grid[row][col] = -1
    }
    
    for _, guard := range guards {
        queue := make([][]int, 0)
        queue = append(queue, guard)
        directions := [][]int{{-1, 0}, {1, 0}, {0, -1}, {0, 1}}
        
        for len(queue) > 0 {
            curr := queue[0]
            queue = queue[1:]
            row, col := curr[0], curr[1]
            
            if row < 0 || row >= m || col < 0 || col >= n || grid[row][col] == -1 || grid[row][col] == 1 {
                continue
            }
            
            grid[row][col] = 1
            
            for _, dir := range directions {
                newRow, newCol := row+dir[0], col+dir[1]
                queue = append(queue, []int{newRow, newCol})
            }
        }
    }
    
    count := 0
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if grid[i][j] == 0 {
                count++
            }
        }
    }
    
    return count
}
